src/ImageKernel.java

```diff
@@ -0,0 +1,49 @@
+public interface ImageKernel extends Image {
```

Comment:
I think you have your relationships backwards. It should be that Image inherits ImageKernel. Then, make sure ImageKernel extends Standard.